packages/@lwc/ssr-runtime/src/index.ts

```diff
@@ -0,0 +1,161 @@
+        const set = new Set(classNameStr.split(MULTI_SPACE).filter(Boolean));
+        if (!set.has(classNameToToggle) && force !== false) {
+            set.add(classNameToToggle);
+        } else if (set.has(classNameToToggle) && force !== true) {
+            set.delete(classNameToToggle);
+        }
+        this.el.className = Array.from(set).join(' ');
+        return set.has(classNameToToggle);
+    }
+}
+
+export class LightningElement {
+    isConnected = false;
+    className = '';
+    // TODO [W-14977927]: protect internals from userland
+    __attrs?: Record<string, string>;
+    __classList: ClassList | null = null;
+
+    constructor(propsAvailableAtConstruction: Record<string, any>) {
+        Object.assign(this, propsAvailableAtConstruction);
+    }
+
+    // TODO [W-14977927]: protect internals from userland
+    __internal__setState(
+        props: Record<string, any>,
+        reflectedProps: string[],
+        attrs: Record<string, any>
+    ) {
+        Object.assign(this, props);
+        this.__attrs = attrs;
+
+        // Whenever a reflected prop changes, we'll update the original props object
+        // that was passed in. That'll be referenced when the attrs are rendered later.
+        for (const reflectedPropName of reflectedProps) {
+            Object.defineProperty(this, reflectedPropName, {
+                get() {
+                    return props[reflectedPropName];
+                },
+                set(newValue) {
+                    props[reflectedPropName] = newValue;
+                },
+                enumerable: true,
+            });
+        }
+
+        Object.defineProperty(this, 'className', {
+            get() {
+                return props.class ?? '';
+            },
+            set(newVal) {
+                props.class = newVal;
+                attrs.class = newVal;
+            },
+        });
+    }
+
+    get classList() {
+        if (this.__classList) {
+            return this.__classList;
+        }
+        return (this.__classList = new ClassList(this));
+    }
+
+    getAttribute(attrName: string) {
+        return this.__attrs?.[attrName] ?? null;
+    }
+}
+
+const escapeAttrVal = (attrVal: string) =>
+    attrVal.replaceAll('&', '&amp;').replaceAll('"', '&quot;');
+
+export function* renderAttrs(attrs: Attributes) {
+    if (!attrs) {
+        return;
+    }
+    for (const [key, val] of Object.entries(attrs)) {
+        if (val) {
+            if (typeof val === 'string') {
+                yield ` ${key}="${escapeAttrVal(val)}"`;
+            } else {
+                yield ` ${key}`;
+            }
+        }
+    }
+}
+
+export function* fallbackTmpl(
+    _props: unknown,
+    _attrs: unknown,
+    _slotted: unknown,
+    Cmp: LightningElementConstructor,
+    _instance: unknown
+) {
+    if (Cmp.renderMode !== 'light') {
+        yield '<template shadowrootmode="open">';
+    }
+    if (Cmp.renderMode !== 'light') {
+        yield '</template>';
+    }
```

Comment:
It cannot. Good catch. There used to be more in this function that got refactored away. And really it can just be a single `yield` statement now, which is even better.